### PR TITLE
Fix Rest Tests Failing to Cleanup Rollup Jobs

### LINKED ISSUE #50819: BWC test failures for MultiVersionRepositoryAccessIT
Our most recent `master` BWC test run failed against versions 7.0.x and 7.1.x with the same test failures indicating we either made a BWC incompatible change in `master` or perhaps just the test needs updating.

The build run in question included [this commit](https://github.com/elastic/elasticsearch/commit/ee6fbcc4a7a79bd7bd3e05768824798af264a30e), which _might_ potentially be related. @original-brownbear any thoughts?

Example build scan: https://gradle-enterprise.elastic.co/s/dpoop7fvn2rwq/tests/failed
Jenkins build with links to each BWC version: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+bwc/471/

### PR DESCRIPTION
**Pull request by original-brownbear**
If the rollup jobs index doesn't exist for some reason (like running against a 6.x cluster)
we should just assume the jobs have been cleaned up and move on.

Closes #50819

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java [hunk 1]
```diff
@@ -663,7 +663,15 @@ private void wipeClusterSettings() throws IOException {
...
 
     private void wipeRollupJobs() throws IOException {
-        Response response = adminClient().performRequest(new Request("GET", "/_rollup/job/_all"));
+        final Response response;
+        try {
+            response = adminClient().performRequest(new Request("GET", "/_rollup/job/_all"));
+        } catch (ResponseException e) {
+            if (e.getResponse().getStatusLine().getStatusCode() == RestStatus.NOT_FOUND.getStatus()) {
+                return;
+            }
```
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java]: I think it is worth adding a comment here explaining why this is ok.

**Review by nik9000 — APPROVED** [review]
Lgtm. I just ask for a comment.

**Review by dakrone — APPROVED** [review]
LGTM also

**Comment by original-brownbear** [comment]
Jenkins run 
elasticsearch-ci/1

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/1 (unrelated failure)

**Comment by original-brownbear** [comment]
Thanks Nik + Lee!

**Comment by polyfractal** [comment]
Heya everyone :)  Found this PRwhile investigating: https://github.com/elastic/elasticsearch/issues/51141

I'm not sure this is the correct approach?  E.g. if that endpoint is failing, it means we're talking to 6.x and need to use the deprecated URLs instead?  Note that `/_rollup/job/_all` is an API endpoint, not an index.  We need to fall back to `/_xpack/rollup/job/_all` for older clusters

When this was fixed in the FullClusterRestart tests, we checked if it was running against the old cluster and [changed URL accordingly](https://github.com/elastic/elasticsearch/pull/36451/files#diff-55c485ebf061c91a72f2a9956fd22aa4R650).

I don't think there is similar ability in ESRestTestCase so I'm not sure the best course of action, but I'm worried that just ignoring cleanup won't work because other tests may run into an existing job that should have been cleaned, but wasn't because the `wipeRollupJob()` method bailed early?

**Comment by original-brownbear** [comment]
Urgh, sorry about that @polyfractal!

We could use a construct like 

```
        if (nodeVersions.stream().allMatch(version -> version.onOrAfter(Version.V_7_6_0))) {
```

in the rest tests and change the URL according to the ES version we're dealing with right?

**Comment by polyfractal** [comment]
No worries!  I'm only vaguely aware of how all these BWC tests work/interact so wasnt sure myself either how to deal with it :)

That sounds like it would work to me!  Do the ESRestTestCases fail if a deprecation warning is thrown?  E.g. if we switch to using the old endpoint because one of the nodes is still 6.x, but talk to a 7.x node, it will throw a deprecation warning.  I remember elsewhere it causing problems because the deprecation failed the test... not sure if that would happen here though

**Pull request merged**